esti/system_test.go

```diff
@@ -150,21 +151,81 @@ func uploadFileRandomDataAndReport(ctx context.Context, repo, branch, objPath st
 }
 
 func uploadFileAndReport(ctx context.Context, repo, branch, objPath, objContent string, direct bool) (checksum string, err error) {
+	// Upload using direct access
 	if direct {
-		stats, err := helpers.ClientUploadDirect(ctx, client, repo, branch, objPath, nil, "", strings.NewReader(objContent))
+		stats, err := uploadContentDirect(ctx, client, repo, branch, objPath, nil, "", strings.NewReader(objContent))
 		if err != nil {
 			return "", err
 		}
 		return stats.Checksum, nil
-	} else {
-		resp, err := uploadContent(ctx, repo, branch, objPath, objContent)
+	}
+	// Upload using API
+	resp, err := uploadContent(ctx, repo, branch, objPath, objContent)
+	if err != nil {
+		return "", err
+	}
+	if err := verifyResponse(resp.HTTPResponse, resp.Body); err != nil {
+		return "", err
+	}
+	return resp.JSON201.Checksum, nil
+}
+
+// uploadContentDirect uploads contents as a file using client-side ("direct") access to underlying storage.
+// It requires credentials both to lakeFS and to underlying storage, but considerably reduces the load on the lakeFS
+// server.
+func uploadContentDirect(ctx context.Context, client apigen.ClientWithResponsesInterface, repoID, branchID, objPath string, metadata map[string]string, contentType string, contents io.ReadSeeker) (*apigen.ObjectStats, error) {
```

Comment:
I'm not sure I understand why we still need this

Reply:
If we do not require direct upload for GC testing, I'll remove this functionality.